pkg/controller/podautoscaler/horizontal.go

```diff
@@ -146,10 +143,65 @@ func (a *HorizontalController) Run(stopCh <-chan struct{}) {
 		return
 	}
 
+	// start a single worker (we may wish to start more in the future
+	go wait.Until(a.worker, time.Second, stopCh)
```

Comment:
Both of these methods wait for the channel to be closed. They neither write to it nor close it.